clustered/integration-test/src/test/java/org/ehcache/clustered/ReconnectDuringDestroyTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered;
+
+import org.ehcache.PersistentCacheManager;
+import org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder;
+import org.ehcache.clustered.client.config.builders.ServerSideConfigurationBuilder;
+import org.ehcache.clustered.client.internal.ClusterTierManagerClientEntity;
+import org.ehcache.clustered.client.internal.lock.VoltronReadWriteLock;
+import org.ehcache.clustered.client.service.EntityBusyException;
+import org.ehcache.clustered.common.EhcacheEntityVersion;
+import org.ehcache.clustered.common.internal.ClusterTierManagerConfiguration;
+import org.ehcache.clustered.util.TCPProxyUtil;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.terracotta.connection.Connection;
+import org.terracotta.connection.entity.EntityRef;
+import org.terracotta.exception.EntityNotFoundException;
+import org.terracotta.testing.rules.Cluster;
+
+import com.tc.net.proxy.TCPProxy;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.ehcache.clustered.reconnect.BasicCacheReconnectTest.RESOURCE_CONFIG;
+import static org.ehcache.clustered.util.TCPProxyUtil.setDelay;
+import static org.terracotta.testing.rules.BasicExternalClusterBuilder.newCluster;
+
+/**
+ * ReconnectDuringDestroyTest
```

Comment:
Document what this test is trying to do as this involves a lot of mimicking of internal flow. It could even be comments within the test code.